lib/src/refresh_when_network_available_extension.dart

```diff
@@ -35,6 +37,17 @@ extension RefreshWhenNetworkAvailableExtension<T>
   /// }
   /// ```
   void refreshWhenNetworkAvailable() {
+    assert(
+      // ignore: deprecated_member_use, ok for riverpod v2
+      this is FutureProviderRef,
+      'refreshWhenNetworkAvailable can only be used on '
+      'FutureProviderRef',
+    );
+    // ignore: deprecated_member_use, ok for riverpod v2
+    if (this is! FutureProviderRef) return;
+    // ignore: deprecated_member_use, ok for riverpod v2
+    final ref = this as FutureProviderRef;
```

Comment:
💅 Replace with a static type check on a local variable holding `this` (like:
```dart
final that = this;

if(thas is! FutureProviderRef) return;
```
)